### PR TITLE
Empty links should not be errors (warnings only)

### LINKED ISSUE #276: Empty links should not be errors
It should be warnings.

Follow-up of https://github.com/openzim/nautilus/issues/35

### PR DESCRIPTION
**Pull request by BartlomiejZalas**
Fixes #276.

- new TestType `URL_EMPTY` introduced (with warning severity)
- MesId `EMPTY_LINKS` moved from test type `URL_INTERNAL` to `URL_EMPTY`
- Tests adapted to new implementation

Tool usage output:
image

### CODE REVIEW

File: src/zimcheck/checks.cpp [hunk 1]
```diff
@@ -70,7 +71,7 @@ std::unordered_map<MsgId, MsgInfo> msgTable = {
   { MsgId::MAIN_PAGE,        { TestType::MAIN_PAGE, "Main Page Index stored in Archive Header: {{&main_page_index}}" } },
   { MsgId::EMPTY_ENTRY,      { TestType::EMPTY, "Entry {{&path}} is empty" } },
   { MsgId::OUTOFBOUNDS_LINK, { TestType::URL_INTERNAL, "{{&link}} is out of bounds. Article: {{&path}}" } },
-  { MsgId::EMPTY_LINKS,      { TestType::URL_INTERNAL, "Found {{&count}} empty links in article: {{&path}}" } },
+  { MsgId::EMPTY_LINKS,      { TestType::URL_EMPTY, "Found {{&count}} empty links in article: {{&path}}" } },
```
Comment (mgautierfr) [inline src/zimcheck/checks.cpp:74]: Please move this line up or down to regroup the `URL_INTERNAL` checks.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
Thanks for your PR.

Just a very small change for cosmetic and we are good.

File: src/zimcheck/checks.cpp (hunk 1 above)
Comment (BartlomiejZalas) [reply src/zimcheck/checks.cpp]: Moved below `URL_EXTERNAL`

**Comment by kelson42** [comment]
@BartlomiejZalas Thank you for your PR and last commit. Looks good, but might take a bit more than a week for the next review pass and merge. Thank you for your patience.

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**